modules/core/src/main/scala/iota/Cop.scala

```diff
@@ -68,6 +68,12 @@ object Cop {
       else None
     def apply(a: A): Cop[L] = inj(a)
     def unapply(c: Cop[L]): Option[A] = proj(c)
+
+    def fold(c: Cop[L]): Either[Cop[TList.Op.Remove[A, L]], A] =
```

Comment:
I was going to generalize the operations into a trait soon, so that they are only defined once and then just mixed in.